q2_amr/tests/card/test_utils.py

```diff
@@ -85,37 +90,36 @@ def test_exception_raised(self):
             self.assertEqual(str(cm.exception), expected_message)
 
     def test_read_in_txt_mags(self):
-        path = self.get_data_path("output.mags.txt")
-        self.read_in_txt_test_body(
-            path, "ARO", "sample1", self.mags_mapping_data_sample1
-        )
-
-    def test_read_in_txt_allele(self):
-        path = self.get_data_path("output.allele_mapping_data.txt")
+        # Test read_read_in_txt with output data from annotate_mags_card
         self.read_in_txt_test_body(
-            path, "ARO Accession", "sample1", self.mapping_data_sample1
+            "output.mags.txt", "sample1", self.count_df_list[2], "mags"
         )
 
-    def test_read_in_txt_gene(self):
-        path = self.get_data_path("output.gene_mapping_data.txt")
+    def test_read_in_txt_reads(self):
+        # Test read_read_in_txt with output data from annotate_reads_card
         self.read_in_txt_test_body(
-            path, "ARO Accession", "sample1", self.mapping_data_sample1
+            "output.allele_mapping_data.txt", "sample1", self.count_df_list[0], "reads"
         )
 
-    def read_in_txt_test_body(self, path, col_name, samp_bin_name, mapping_data):
+    def read_in_txt_test_body(self, txt_file, samp_bin_name, mapping_data, data_type):
+        # Create expected and observed count dataframes and compares them
         exp = mapping_data
-        obs = read_in_txt(path, col_name, samp_bin_name)
-        obs[col_name] = obs[col_name].astype(int)
+        obs = read_in_txt(self.get_data_path(txt_file), samp_bin_name, data_type)
         pd.testing.assert_frame_equal(exp, obs)
 
     def test_create_count_table(self):
-        df_list = [self.mapping_data_sample1, self.mapping_data_sample2]
+        # Create observed count table with create_count_table function
+        df_list = [self.count_df_list[0], self.count_df_list[1]]
         obs = create_count_table(df_list)
-        mag_test_class = TestAnnotateMagsCard()
-        exp = mag_test_class.table
+        obs = obs.astype(str)
+
+        # Create expected count table from
+        exp = self.frequency_table
         exp.set_index("sample_id", inplace=True)
```

Comment:
I would move this statement to where the actual df is being created - if you do it like this, it'll update it in place _here_ but will have an effect in every other test which may be using this df so it would be a bit difficult to debug in case something goes wrong.